frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -292,15 +292,13 @@ export default ({
             const msgValue = JSON.parse(message.opValue())
             const { meta, data } = msgValue
             this.messageState.contract.messages.push({
-              // TODO: message.hash() could be different from this message on event listener
-              // https://github.com/okTurtles/group-income/issues/1487
-              // so that the message.hash() can't be used as the identifier of this message
-              // https://github.com/okTurtles/group-income/issues/1503
-              // After this issue is fixed, we can fix this TODO and identity messages using message.id() instead of hash()
               ...createMessage({ meta, data, hash: message.hash() }),
-              // TODO: pending is useful to turn the message gray meaning failed (just like Slack)
+              // NOTE: pending is useful to turn the message gray meaning failed (just like Slack)
               // when we don't get event after a certain period
-              pending: true
+              pending: true, // NOTE: do not RENAME this as it's being used inside the contract
+              // message.id() should be used as an identifier of GIMessage
+              // https://github.com/okTurtles/group-income/issues/1503
+              giMsgID: message.id() // NOTE: do not RENAME this as it's being used inside the contract
```

Comment:
I think this can be deleted once `id` is passed in to `createMessage` :)

Reply:
Correct.